src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -407,18 +407,54 @@ bool P104_data_struct::handlePluginWrite(taskIndex_t   taskIndex,
   return false;       // Default: unknown command
 }
 
-void P104_data_struct::getTime(char *psz,
-                               bool  f = true) {
-  uint16_t h, m;
+void getTime(char *psz,
+             bool  seconds = false,
+             bool  colon   = true) {
+  uint16_t h, m, s;
 
   h = node_time.hour();
   m = node_time.minute();
-  sprintf(psz, "%02d%c%02d", h, (f ? ':' : ' '), m);
+
+  if (!seconds) {
+    sprintf(psz, "%02d%c%02d", h, (colon ? ':' : ' '), m);
+  } else {
+    s = node_time.second();
+    sprintf(psz, "%02d%c%02d %02d", h, (colon ? ':' : ' '), m, s);
+  }
+}
+
+void getDate(char *psz,
+             bool  year    = true,
+             bool  fourDgt = false) {
+  uint16_t d, m, y;
+
+  d = node_time.day();
+  m = node_time.month();
+
+  if (year) {
+    y = node_time.year() - (fourDgt ? 0 : 2000); // Will we survive into 21xx?
```

Comment:
Past 03:14:07 UTC on 19 January 2038 is already becoming an issue I'm afraid. ;)

Reply:
By that time I should be retired 🤣 

On Topic: We might want to fix that `node_time` issue somewhere before that date then.